cfme/tests/physical_infrastructure/ui/test_physical_server_list_buttons.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+import pytest
+from selenium.common.exceptions import StaleElementReferenceException
+
+from cfme.physical.provider.lenovo import LenovoProvider
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.common.physical_server_views import (
+    PhysicalServersView,
+)
+from cfme.utils.wait import wait_for
+
+pytestmark = [pytest.mark.tier(3), pytest.mark.provider([LenovoProvider], scope="module")]
+
+
+@pytest.fixture(scope="module")
+def physical_servers_collection(appliance, setup_provider_modscope):
+    # Get and return the first physical server
+    return appliance.collections.physical_servers
+
+
+@pytest.fixture(scope="module")
+def physical_servers(appliance, provider):
+    # Get and return the first physical server
+    return appliance.collections.physical_servers.all(provider)
+
+
+# Configuration Button
+def test_refresh_relationships(physical_servers_collection, provider, physical_servers):
+    last_refresh = provider.last_refresh_date()
+    physical_servers_collection.refresh(provider, physical_servers[0], physical_servers[1])
+    assert last_refresh != provider.last_refresh_date()
+
+
+actions = [
+    "power_off",
+    "power_on",
+    "power_off_now",
+    "restart",
+    "restart_now",
+    "restart_to_sys_setup",
+    "restart_to_sys_setup",
+    "blink_loc_led",
+    "turn_off_loc_led",
+    "turn_on_loc_led",
+]
+
+
+# Power / Identify Buttons
+@pytest.mark.parametrize("action", actions)
+def test_server_actions(physical_servers_collection, provider, physical_servers, action):
+    """ Test the physical server actions sending the action request, waiting the task be complete on MiQ
+        and then waiting for the state of some attribute of the physical server be changed
+    Params:
+        * action: the action to be performed against the Physical Server
+    Metadata:
+        test_flag: crud
+    """
+    view = provider.create_view(PhysicalServersView)
+    message = 'Requested Server {} for the selected servers'.format(action)
+    getattr(physical_servers_collection, action)(physical_servers[0], physical_servers[1])
```

Comment:
This can generate IndexErrors if there aren't actually two physical servers available and the list only contains a single item.

Per my comment above, I think the test would be more consistent and deterministic if a fixture actually asserts the number of available physical servers on the provider, skipping the test if there aren't at least two, as the test requires that many.

Reply:
yap, I agreed with you. thanks.